### PR TITLE
8295792: Clean up old async close code

### PR DESCRIPTION
**Pull request by djelinski**
Please review this PR that removes the remains of old fdTable-based socket close synchronization.

Verified that tier1-3 tests continue to pass on Linux, MacOS and Windows. Did not verify AIX.

Also removed one slow test for an issue in fdtable implementation; as far as I could tell, it is no longer relevant.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8295792](https://bugs.openjdk.org/browse/JDK-8295792): Clean up old async close code

### Reviewers
 * [Alan Bateman](https://openjdk.org/census#alanb) (@AlanBateman - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk pull/10816/head:pull/10816` \
`$ git checkout pull/10816`

Update a local copy of the PR: \
`$ git checkout pull/10816` \
`$ git pull https://git.openjdk.org/jdk pull/10816/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 10816`

View PR using the GUI difftool: \
`$ git pr show -t 10816`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/10816.diff">https://git.openjdk.org/jdk/pull/10816.diff</a>

</details>

### CODE REVIEW

File: src/java.base/unix/native/libnet/net_util_md.c [hunk 1]
```diff
@@ -715,7 +700,7 @@ NET_Wait(JNIEnv *env, jint fd, jint flags, jint timeout)
           pfd.events |= POLLOUT;
 
         errno = 0;
-        read_rv = NET_Poll(&pfd, 1, nanoTimeout / NET_NSEC_PER_MSEC);
+        read_rv = poll(&pfd, 1, nanoTimeout / NET_NSEC_PER_MSEC);
```
Comment (AlanBateman) [inline src/java.base/unix/native/libnet/net_util_md.c:703]: I suspect this one will need to handle EINTR like the original code.

Comment (djelinski) [reply src/java.base/unix/native/libnet/net_util_md.c:703]: as far as I can tell, this method retries all poll failures regardless of errno until timeout expires; NET_Poll retried EINTR, so I think we're good here. Or am I missing something?

Comment (AlanBateman) [reply src/java.base/unix/native/libnet/net_util_md.c:703]: I think you are right, it's ignoring all errors so will continue to loop if interrupted.

**Review by AlanBateman — APPROVED** [review]
Good cleanup. At some point I think we should put error handling into NET_Wait but not here.

**Pull request closed without merging**